src/main/java/com/hermesworld/ais/galapagos/topics/controller/TopicController.java

```diff
@@ -143,19 +143,22 @@ public void addProducerToTopic(@PathVariable String environmentId, @PathVariable
 
     }
 
-    @PostMapping(value = "/api/change-owner/{envId}/{topicName}/{producerApplicationId}")
+    @PostMapping(value = "/api/change-owner/{envId}/{topicName}")
     public void changeTopicOwner(@PathVariable String envId, @PathVariable String topicName,
-            @PathVariable String producerApplicationId) {
-        if (envId.isEmpty() || topicName.isEmpty() || producerApplicationId.isEmpty()) {
+            @RequestBody ChangeTopicOwnerDto request) {
+        if (envId.isEmpty() || topicName.isEmpty() || request.getProducerApplicationId().isEmpty()) {
             throw new ResponseStatusException(HttpStatus.BAD_REQUEST);
         }
 
-        if (!applicationsService.isUserAuthorizedFor(producerApplicationId)) {
+        TopicMetadata topic = topicService.getTopic(envId, topicName).orElseThrow(notFound);
+        if (applicationsService.getUserApplicationOwnerRequests().stream()
+                .noneMatch(req -> req.getState() == RequestState.APPROVED
+                        && topic.getOwnerApplicationId().equals(req.getApplicationId()))) {
```

Comment:
Why not just `applicationsService.isUserAuthorizedFor(topic.getOwnerApplicationId())`?